File: src/main/java/trop/TROPCreativeTabs.java
package trop;

import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.*;
import net.minecraftforge.event.CreativeModeTabEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.registries.RegistryObject;

public class TROPCreativeTabs {
	@SubscribeEvent
	public static void addCreativeTab(CreativeModeTabEvent.Register event) {
		event.registerCreativeModeTab(new ResourceLocation("trop", "troptab"), builder -> builder.title(Component.translatable("itemGroup.ringPower")).icon(() -> new ItemStack(TROP.ring_jiindur.get())).displayItems((enabledFlags, populator, hasPermissions) -> {
			for (RegistryObject<Item> amogus : TROP.sus) {
				populator.accept(amogus.get());
			}
		}));
	}
}

File: kotlin/src/main/java/trop/TROPCreativeTabs.kt
package trop

import net.minecraft.network.chat.Component
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.ItemStack
import net.minecraftforge.event.CreativeModeTabEvent.Register
import net.minecraftforge.eventbus.api.SubscribeEvent

object TROPCreativeTabs {
	@SubscribeEvent
	fun addCreativeTab(event: Register) {
		event.registerCreativeModeTab(
			ResourceLocation("trop", "troptab")
		) {
			it.title(Component.translatable("itemGroup.ringPower")).icon {
				ItemStack(TROP.ring_jiindur.get())
			}.displayItems { _, populator ->
				for (item in TROP.itemList) {
					populator.accept(item)
				}
			}
		}
	}
}

